src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -1665,10 +1665,10 @@ private Body.Builder visitCaseLabel(JCTree tree, JCExpression selector, Value ta
             return body;
         }
 
-        private Body.Builder visitCaseBody(JCTree.JCCase c, FunctionType caseBodyType) {
+        private Body.Builder visitCaseBody(JCTree tree, JCTree.JCCase c, FunctionType caseBodyType) {
 
             Body.Builder body = null;
-            Type yieldType = typeElementToType(caseBodyType.returnType());
+            Type yieldType = tree.type != null ? adaptBottom(tree.type) : null;
```

Comment:
Why the change here?

Reply:
`typeElementToType(javaType.STRING)` returns `none`

Reply:
I need to update `primitiveAndBoxTypeMap`

Reply:
Uhm - I see. `typeElementToType` only works for simple types - but in this case I believe the case body can return any type - e.g. with patterns you can switch on any reference type.

Reply:
yes, switch expression can yield a result of any type.

Reply:
I think the last PR unifying code on `ReflectMethods` introduce a regression causing failures in `NullTest`. This should should fix it.